tests/integration/proxy/testproxymain.cpp

```diff
@@ -758,17 +778,15 @@ int main() {
 
     Glib::RefPtr<Glib::MainLoop> ml = Glib::MainLoop::create();
 
+    TestProxyImpl proxyTest;
+    proxyTest.done().connect([&]() { ml->quit(); });
```

Comment:
Note that important... but would it be less code to just pass the main loop to TestProxyImpl(), have it as an instance variable and call the quit() method instead of adding a done() signal?

Reply:
Probably, but I think that having a done() signal is more clear, and a mechanism someone might want to reuse in production code.